File: python/threaded_csv/find.py
#!/usr/bin/python

import os, sys

def find_item(d):
	try:
		files = os.listdir(d)
	except OSError, err:
		print(d+": "+str(err))
	else:
		print(d)
		for item in files:
			fullitem = os.path.join(d, item)
			if (os.path.isdir(fullitem)):
				find_item(fullitem)
			else:
				print(fullitem)

if (len(sys.argv)>1):
	s_dir = sys.argv[1]
else:
	s_dir = "./"
find_item(s_dir)

File: perl/find.pl
#!/usr/bin/perl

use strict;
use warnings;
use File::Spec;

sub find_item{
	my $dir=File::Spec->canonpath($_[0]);
	my $DIR;
	print("$dir\n");
	eval{
		opendir($DIR,$dir);
	};
	if ($@){
		print "$@\n";
	}
	else{
		while (my $item=readdir($DIR)){
			if ($item ne "." and $item ne ".."){
				my $fullitem=File::Spec->join($dir,$item);
				if (-d $fullitem){
					find_item($fullitem);
				}
				else{
					print "$fullitem\n";
				}
			}
		}
		closedir($DIR);
	}
}

my $s_dir=$ARGV[0];
find_item($s_dir);

